health/metrics/plugin_impl_metrics.go

```diff
@@ -0,0 +1,234 @@
+	dependencyLabel    string = "dependency"
+	buildVersionLabel  string = "build_version"
+	buildDateLabel     string = "build_date"
+	serviceHealthName  string = "service_health"
+	// Adapt Ligato status code for now.
+	// TODO: Consolidate with that from the "Common Container Telemetry" proposal.
+	//serviceHealthHelp    string = "The health of the serviceLabel 0 = INIT, 1 = UP, 2 = DOWN, 3 = OUTAGE"
+	serviceHealthHelp    string = "The health of the serviceLabel 0 = INIT, 1 = OK, 2 = ERROR"
+	dependencyHealthName string = "service_dependency_health"
+	// Adapt Ligato status code for now.
+	// TODO: Consolidate with that from the "Common Container Telemetry" proposal.
+	//dependencyHealthHelp string = "The health of the dependencyLabel 0 = INIT, 1 = UP, 2 = DOWN, 3 = OUTAGE"
+	dependencyHealthHelp string = "The health of the dependencyLabel 0 = INIT, 1 = OK, 2 = ERROR"
+	serviceInfoName      string = "service_info"
+	serviceInfoHelp      string = "Build info for the service.  Value is always 1, build info is in the tags."
+)
+
+var (
+	httpPort string
+)
+
+// init is here only for parsing program arguments
+func init() {
+	flag.StringVar(&httpPort, "prometheus-http-port", rest.DefaultHTTPPort,
+		"Listening port for the Agent's Prometheus health/metrics port.")
+}
+
+// Plugin struct holds all plugin-related data.
+type Plugin struct {
+	Deps
+
+	customProbe bool
+}
+
+// Deps lists dependencies of the Prometheus plugin.
+type Deps struct {
+	local.PluginLogDeps                               // inject
+	HTTP                *rest.Plugin                  // inject (optional)
+	StatusCheck         statuscheck.AgentStatusReader // inject
+}
+
+// Init may create a new (custom) instance of HTTP if the injected instance uses
+// different HTTP port than requested.
+func (p *Plugin) Init() (err error) {
+	// Start Init() and AfterInit() for new Prometheus in case the port is different
+	// from agent http.
+	if p.HTTP.HTTPport != httpPort {
+		childPlugNameHTTP := p.String() + "-HTTP"
+		p.HTTP = &rest.Plugin{
+			Deps: rest.Deps{
+				Log:        logging.ForPlugin(childPlugNameHTTP, p.Log),
+				PluginName: core.PluginName(childPlugNameHTTP),
+				HTTPport:   httpPort,
+			},
+		}
+		err := p.HTTP.Init()
+		if err != nil {
+			return err
+		}
+		err = p.HTTP.AfterInit()
+		if err != nil {
+			return err
+		}
+
+		p.customProbe = true
+	}
+
+	p.RegisterGauge(
+		namespace,
+		subsystem,
+		serviceHealthName,
+		serviceHealthHelp,
+		prometheus.Labels{serviceLabel: agentName},
+		p.getServiceHealth,
+	)
+
+	agentStatus := p.StatusCheck.GetAgentStatus()
+	p.RegisterGauge(
+		namespace,
+		subsystem,
+		serviceInfoName,
+		serviceInfoHelp,
+		prometheus.Labels{
+			serviceLabel:      agentName,
+			buildVersionLabel: agentStatus.BuildVersion,
+			buildDateLabel:    agentStatus.BuildDate},
+		func() float64 { return 1 },
+	)
+
+	return nil
+}
+
+// AfterInit registers HTTP handlers.
+func (p *Plugin) AfterInit() error {
+	if p.HTTP != nil {
```

Comment:
We will need to extract the code (form both probes plugin) that create yet another instance of HTTP server into some helper function and use it in rpc flavor.

Reply:
later?  since probe plugin has the same code now :-)